modules/apps/remote-app/remote-app-support-web/src/main/resources/META-INF/resources/index.js

```diff
@@ -123,6 +123,20 @@ function receiveMessage(event) {
 					const resource = data.resource;
 					const init = data.init;
 
+					const {body} = init;
+
+					if (body.__FORM_DATA__) {
+						const formData = new FormData();
+
+						Object.entries(body.__FORM_DATA__).forEach(
+							(key, value) => {
```

Comment:
`Object.entries` returns an array of tuples so this should be:

```
([key value]) => {
```

(Note the destructuring.)

Did you test this? Because without the destructuring, `(key, value)` is going to take on these values:

```
- `key` = `[key0, value0]`, `value` = `0`
- `key` = `[key1, value1]`, `value` = `1`
- etc
````

Reply:
Another thing I've forgotten more than once, but I did try it, which makes me wonder how it could have worked.
Going to take another look